src/main/java/com/userservice/controller/UserController.java

```diff
@@ -0,0 +1,61 @@
+package com.userservice.controller;
+
+import com.userservice.exception.UserExistsException;
+import com.userservice.model.User;
+import com.userservice.service.UserService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+
+@RestController
+@RequestMapping("/v1")
+public class UserController {
+
+    private static Logger logger = LoggerFactory.getLogger(UserController.class);
+    private UserService userService;
+
+    public UserController(UserService userService) {
```

Comment:
Why is there no @Autowiring here ? how is it working ?

Reply:
Starting with Spring 4.3, if a class, which is configured as a Spring bean, has only one constructor, the @Autowired annotation can be omitted and Spring will use that constructor and inject all necessary dependencies.

Regarding the default constructor: You either need the default constructor, a constructor with the @Autowired annotation when you have multiple constructors, or only one constructor in your class with or without the @Autowired annotation.